chocolatey/plugins/module_utils/Packages.psm1

```diff
@@ -190,7 +190,17 @@ function Get-ChocolateyVersion {
 
     # Prerelease versions are not relevant for our purposes.
     # Stripping off any prerelease tag here gets us enough for what we need.
-    ($script:ChocolateyVersion = [version]($result.stdout -replace '-.+$'))
+    # Also, if a license is installed, but the licensed extension is missing,
+    # choco.exe output will contain an error which we need to ignore for the
+    # purposes of determining the version of Chocolatey CLI.
+    $SemVerRegex = '(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?'
```

Comment:
I'm torn on whether we use the recommended regex as-is or adapt for PowerShell which uses case-insensitive matching by default...

```suggestion
    $SemVerRegex = '(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-z-][0-9a-z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-z-][0-9a-z-]*))*))?(?:\+([0-9a-z-]+(?:\.[0-9a-z-]+)*))?'
```

Reply:
I'm ashamed to say that I have a lot of regex all through my PowerShell code with unnecessary case-sensitive matching because I always forget that `-match` is save-insensitive 😅

My gut says use the recommended one as is. Makes it easier to trace back to the source and if there's any update to it in the future then we don't need to consider whether we change it again when adopting the update.

Reply:
Good thinking, I've just added that in a second commit (which will trigger the pipeline again 😅)